### PR TITLE
Buildsystem: the build system now correctly assigns the hash, tag, or version when gathering build information

### PR DESCRIPTION
**Pull request by fsareshwala**

### CODE REVIEW

File: src/cc/common/buildversgit.sh [hunk 1]
```diff
@@ -20,60 +11,178 @@
...
-    exit
-fi
+usage() {
+    echo "
+usage: $0 <options>
+
+    Options:
+        --info BINARY               get build information from the binary
+        --head                      print the head hash of qfs
+        --release                   print the release verison or hash of qfs
```
Comment (mckurt) [inline src/cc/common/buildversgit.sh]: version

File: src/cc/common/buildversgit.sh [hunk 2]
```diff
@@ -20,60 +11,178 @@
...
+}
 
-# Set official release version here.
-qfs_release_version="1.1.4"
-qfs_source_revision=""
+get_head() {
+    if ! git="$(command -v git)" || [ -z "$git" ]; then
+        echo "unknown"
+        return 0
+    fi
```
Comment (mckurt) [inline src/cc/common/buildversgit.sh]: "if ! git="$(command -v git)" || [ -z "$git" ]; then
       echo "unknown"
       return 0
fi"

We can make this a function and make calls to it. It's used almost in all functions.

File: src/cc/common/buildversgit.sh (hunk 1 above)
Comment (fsareshwala) [inline src/cc/common/buildversgit.sh]: Fixed.

File: src/cc/common/buildversgit.sh (hunk 2 above)
Comment (fsareshwala) [inline src/cc/common/buildversgit.sh]: Done. Thanks! It actually adds a bit more lines but it encapsulates _how_ we check for git presence.

**Pull request merged**